### PR TITLE
feat(grid): Migrate to 6px baseline grid with 16px standard type

### PR DESCRIPTION
**Pull request by markdalgleish**
## RFC

https://github.com/seek-oss/seek-style-guide/issues/312

## Reason for Change

In order to support a larger standard body copy size of 16px / 24px (instead of 14px / 18px), we've also had to change the underlying baseline grid from 9px to 6px.

## Breaking Changes

- You will need to perform a design audit of your project, since the underlying grid and typography has changed significantly.
- The `superstandard` and `substandard` type variants have been removed from React components and Less variables/mixins. Consumers should use `standard` type instead.

### Design Audit

During the migration process, and at the end as a final review, you should seek guidance from the designer(s) on your team, or those that you collaborate with most often.

The grid and typography changes affect a large portion of the components, variables and mixins provided by this library. Even in cases where the public API of a component doesn't change, its internal dimensions may have changed, which could have flow-on effects for the design of your product.

### Grid Variables

Use the new `@row-height` variable instead of `@grid-row-height`, but be aware that since the value has changed from `9px` to `6px`, **you'll need to revisit any existing calculations in your project.**

For example:

```diff
-@grid-row-height * 2
+@row-height * 3
```

When changing calculations like this, you're encouraged to visually check the result of each change, but you can use this table as a quick migration reference to speed up the process:

<table>
<tr><th>@grid-row-height * X</th><th>@row-height * X</th></tr>
<tr><td>1</td><td>2</td></tr>
<tr><td>2</td><td>3</td></tr>
<tr><td>3</td><td>5</td></tr>
<tr><td>4</td><td>6</td></tr>
<tr><td>5</td><td>7</td></tr>
<tr><td>6</td><td>9</td></tr>
<tr><td>7</td><td>11</td></tr>
</table>

If you arrive at different values in different scenarios, that's totally fine.

For consistency, all grid variables have been updated to follow the same naming convention, but since their values remain the same, you can safely rename all existing usages:

```diff
-@grid-gutter-width
+@gutter-width

-@grid-column-width
+@column-width

-@grid-container-width
+@container-width
```

To reduce breakage of custom layouts, the old grid variables are still provided with their existing values, but **you're encouraged to migrate away from the old variables as soon as possible.** Until you fully migrate all grid variables, your custom layouts might not necessarily be broken, but they won't be following the new grid.

### Typography

All typography has been adjusted to fit the new grid, with most changes simply trying to match the existing design as closely as possible, with some notable exceptions.

**The `superstandard` and `substandard` variants have been removed.** In most cases, you should use standard type instead.

For example, when using the `Text` component:

```diff
-<Text substandard>...
+<Text>...

-<Text superstandard>...
+<Text>...
```

Or, for older projects, when using the Less mixins:

```diff
-.superstandardText()
+.standardText()

-.superstandardTextResponsive()
+.standardTextResponsive()

-.substandardText()
+.standardText()

-.substandardTextResponsive()
+.standardTextResponsive()
```

Even though we encourage the use of larger body copy, **for some existing usages of `substandard`, you may want to keep the text at `14px`.** In these cases, you should switch to `small` text instead:

```diff
-<Text substandard>...
+<Text small>...
```

### CODE REVIEW

File: react/Text/Text.less [hunk 1]
```diff
@@ -15,29 +15,21 @@
...
-      padding: 0 @grid-gutter-width;
+      padding: 0 @gutter-width;
     }
   }
 }
 
 .standard {
   &:not(.baseline) { .standardTextResponsive(@baseline: false); }
   &.baseline { .standardTextResponsive(@baseline: true); }
+  padding-bottom: @row-height;
```
Comment (michaeltaranto) [inline react/Text/Text.less]: This should be added to `small` too i feel

File: react/Autosuggest/Autosuggest.less [hunk 2]
```diff
@@ -39,19 +39,15 @@
 }
 
 .suggestionsContainer_withLabel {
-  @labelRowOffset: 4;
-  top: @grid-row-height * (@suggestionContainerRowsOffset + @labelRowOffset);
-  @media @desktop {
-    @labelRowOffset: 3;
-    top: @grid-row-height * (@suggestionContainerRowsOffset + @labelRowOffset);
-  }
+  @labelRowOffset: 6;
```
Comment (michaeltaranto) [inline react/Autosuggest/Autosuggest.less]: I feel like this works better as `5`. The spacing between the field (when the field has a label) and autosuggest list. At the moment it gives 2 rows, but i prefer 1. Thoughts? (Also the label is 5 rows high now that we have reduced standard copy to only have one spacing row below)

Comment (georgespyropoulos) [reply react/Autosuggest/Autosuggest.less]: Yeah I agree, I think 5 is better

Comment (markdalgleish) [reply react/Autosuggest/Autosuggest.less]: I'm out today, so feel free to update the PR if you like 👌

Comment (michaeltaranto) [reply react/Autosuggest/Autosuggest.less]: Done

**Comment by michaeltaranto** [comment]
Also pushed the optimisation around the descender scale, made this align more perfectly with Roboto, while not disturbing the rhythm in our fallback font stack

File: docs/src/components/App/Header/Header.less [hunk 3]
```diff
@@ -129,8 +129,10 @@
   color: white;
   text-decoration: none;
   display: inline-block;
-  margin-bottom: @grid-row-height;
   &:hover {
     box-shadow: inset 0 -2px 0 @sk-white;
   }
+  @media @desktop {
+    margin-bottom: (@row-height * 2);
+  }
```
Comment (michaeltaranto) [inline docs/src/components/App/Header/Header.less]: This doesnt look right on the docs site, the links in the menu on desktop appear too loose. I'd just remove it.

File: react/Columns/Columns.less [hunk 4]
```diff
@@ -10,15 +10,12 @@
   @media @desktop {
     flex-basis: 0;
     flex-grow: 1;
-
-    min-width: 0;
```
Comment (michaeltaranto) [inline react/Columns/Columns.less]: This `min-width` looks like it is required to support the flexible `Columns` work implemented in https://github.com/seek-oss/seek-style-guide/pull/350

**Comment by markdalgleish** [comment]
@michaeltaranto Updated 👍

File: react/Columns/Columns.less (hunk 4 above)
Comment (markdalgleish) [reply react/Columns/Columns.less]: Nice catch 👏

**Pull request merged**